openvmm/openvmm_entry/src/lib.rs

```diff
@@ -1127,24 +1130,50 @@ fn vm_config_from_command_line(
     }
 
     let mut virtio_devices = Vec::new();
+    let mut add_virtio_device = |bus, resource: Resource<VirtioDeviceHandle>| {
+        let bus = match bus {
+            VirtioBusCli::Auto => {
+                // Use VPCI when possible (currently only on Windows and macOS due
+                // to KVM backend limitations).
+                if with_hv && (cfg!(windows) || cfg!(target_os = "macos")) {
+                    None
+                } else {
+                    Some(VirtioBus::Pci)
```

Comment:
a somewhat spooky choice for auto in the absence of vpci, given that at the moment, we only have a emulated PCI bus for gen1 configurations

Reply:
We also, apparently, have it for Linux direct. And MMIO only works for Linux direct, whereas this works with both gen1 and Linux direct, so....

Reply:
ah, that's right - I recall asking brian to help me with some of the ACPI code to get the gen1 bus tested using linux direct, ha. OK, fair enough!